state/export_test.go

```diff
@@ -580,3 +581,43 @@ func PrimeUnitStatusHistory(
 func GetInternalWorkers(st *State) worker.Worker {
 	return st.workers
 }
+
+// ResourceStoragePath returns the path used to store resource content
+// in the managed blob store, given the resource ID.
+func ResourceStoragePath(c *gc.C, st *State, id string) string {
+	p := NewResourcePersistence(st.newPersistence())
+	_, storagePath, err := p.GetResource(id)
+	c.Assert(err, jc.ErrorIsNil)
+	return storagePath
+}
+
+// IsBlobStored returns true if a given storage path is in used in the
+// managed blob store.
+func IsBlobStored(c *gc.C, st *State, storagePath string) bool {
+	stor := storage.NewStorage(st.ModelUUID(), st.MongoSession())
+	r, _, err := stor.Get(storagePath)
+	if err == nil {
```

Comment:
This reads funny, but maybe only to me...
Would it not be simpler to start with:
...
if err != nil {
    if errors.isNotFound(err) {
        return false
    }
    r.Close()
    return true
}
c.Fatalf(...)
return false
}
...